src/graphing.module.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  SpelunkedEdge,
+  SpelunkedNode,
+  SpelunkedTree,
+} from './spelunker.interface';
+
+export class GraphingModule {
+  static graph(tree: SpelunkedTree[]): SpelunkedNode {
+    const nodeMap = tree.reduce(
+      (map, module) =>
+        map.set(module.name, {
+          dependencies: new Set(),
+          dependents: new Set(),
+          module,
+        }),
+      new Map<string, SpelunkedNode>(),
+    );
+
+    nodeMap.forEach((node) => this.findDependencies(node, nodeMap));
+
+    return this.findRoot(nodeMap);
+  }
+
+  static getEdges(root: SpelunkedNode): SpelunkedEdge[] {
```

Comment:
This is also agnostic to any graphing lib and just returns another generic graph-like data structure the represents an edge between 2 nodes.